fabric-game-rule-api-v1/src/main/java/net/fabricmc/fabric/mixin/gamerule/GameRuleManagementHandlerImplMixin.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.gamerule;
+
+import com.llamalad7.mixinextras.sugar.Local;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.server.dedicated.MinecraftDedicatedServer;
+import net.minecraft.server.dedicated.management.ManagementLogger;
+import net.minecraft.server.dedicated.management.RpcException;
+import net.minecraft.server.dedicated.management.dispatch.GameRuleRpcDispatcher;
+import net.minecraft.server.dedicated.management.handler.GameRuleManagementHandlerImpl;
+import net.minecraft.server.dedicated.management.network.ManagementConnectionId;
+import net.minecraft.world.GameRules;
+
+import net.fabricmc.fabric.api.gamerule.v1.rule.DoubleRule;
+import net.fabricmc.fabric.api.gamerule.v1.rule.EnumRule;
+import net.fabricmc.fabric.impl.gamerule.rpc.FabricGameRuleType;
+import net.fabricmc.fabric.impl.gamerule.rpc.FabricTypedRule;
+
+@Mixin(GameRuleManagementHandlerImpl.class)
+public abstract class GameRuleManagementHandlerImplMixin {
+	@Shadow
+	@Final
+	private MinecraftDedicatedServer server;
+
+	@Shadow
+	public abstract GameRuleRpcDispatcher.TypedRule toTypedRule(String name, GameRules.Rule<?> gameRule);
+
+	@Shadow
+	@Final
+	private ManagementLogger logger;
+
+	@Inject(method = "updateRule", at = @At(value = "INVOKE_ASSIGN", target = "Lnet/minecraft/world/GameRules$Rule;serialize()Ljava/lang/String;"), cancellable = true)
+	private void updateRule(GameRuleRpcDispatcher.UntypedRule untypedRule, ManagementConnectionId remote, CallbackInfoReturnable<GameRuleRpcDispatcher.TypedRule> cir, @Local GameRules.Rule<?> rule, @Local String from) {
```

Comment:
If you are capturing the result of this assignment, switch to ModifyExpressionValue + Cancellable. If not, switch to INVOKE.

Reply:
INVOKE_ASSIGN should basically never be used.

Reply:
Seems WrapOperation would let you avoid all locals, actually.

Reply:
Very clever, yes that seems to work. I didnt know you could cancel wrap operations :) Many thanks.